test/src/test/java/org/apache/accumulo/test/functional/ConcurrentDeleteTableIT.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.accumulo.core.zookeeper.ZooUtil;
+import org.apache.accumulo.fate.AdminUtil;
+import org.apache.accumulo.fate.AdminUtil.FateStatus;
+import org.apache.accumulo.fate.ZooStore;
+import org.apache.accumulo.fate.zookeeper.IZooReaderWriter;
+import org.apache.accumulo.fate.zookeeper.ZooReaderWriter;
+import org.apache.accumulo.harness.AccumuloClusterIT;
+import org.apache.hadoop.io.Text;
+import org.apache.zookeeper.KeeperException;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ConcurrentDeleteTableIT extends AccumuloClusterIT {
+
+  private static final String SCHEME = "digest";
+
+  private static final String USER = "accumulo";
+
+  @Test
+  public void testConcurrentDeleteTablesOps() throws Exception {
+    final Connector c = getConnector();
+    String[] tables = getUniqueNames(2);
+
+    TreeSet<Text> splits = new TreeSet<>();
+
+    for (int i = 0; i < 1000; i++) {
+      Text split = new Text(String.format("%09x", i * 100000));
+      splits.add(split);
+    }
+
+    ExecutorService es = Executors.newFixedThreadPool(20);
+
+    int count = 0;
+    for (final String table : tables) {
+      c.tableOperations().create(table);
+      c.tableOperations().addSplits(table, splits);
+      writeData(c, table);
+      if (count == 1) {
+        c.tableOperations().flush(table, null, null, true);
+      }
+      count++;
+
+      final CountDownLatch cdl = new CountDownLatch(20);
+
+      List<Future<?>> futures = new ArrayList<>();
+
+      for (int i = 0; i < 20; i++) {
+        Future<?> future = es.submit(new Runnable() {
+
+          @Override
+          public void run() {
+            try {
+              cdl.countDown();
+              cdl.await();
+              c.tableOperations().delete(table);
+            } catch (TableNotFoundException e) {
+              // expected
+            } catch (InterruptedException | AccumuloException | AccumuloSecurityException e) {
+              throw new RuntimeException(e);
+            }
+          }
+        });
+
+        futures.add(future);
+      }
+
+      for (Future<?> future : futures) {
+        future.get();
+      }
+
+      try {
+        c.createScanner(table, Authorizations.EMPTY);
+        Assert.fail("Expected table " + table + " to be gone.");
+      } catch (TableNotFoundException tnfe) {
+        // expected
+      }
+
+      FateStatus fateStatus = getFateStatus();
+
+      // ensure there are no dangling locks... before ACCUMULO-4575 was fixed concurrent delete tables could fail and leave dangling locks.
+      Assert.assertEquals(0, fateStatus.getDanglingHeldLocks().size());
+      Assert.assertEquals(0, fateStatus.getDanglingWaitingLocks().size());
+    }
+
+    es.shutdown();
+  }
+
+  private FateStatus getFateStatus() throws KeeperException, InterruptedException {
+    Instance instance = getConnector().getInstance();
+    AdminUtil<String> admin = new AdminUtil<>(false);
+    String secret = "DONTTELL";
```

Comment:
The right place to get it from would be the `AccumuloCluster` available to you from the `getCluster()` static method on AccumuloClusterIT. Sadly, this is not presently exposed.

I think adding a `AccumuloConfiguration getServerConfiguration();` method onto `AccumuloCluster` is the proper fix.

From there, a `MiniAccumuloClusterImpl` can create an `AccumuloConfiguration` with its `accumulo-site.xml` file and a `StandaloneAccumuloCluster` can create one from its `serverAccumuloConfDir` variable.